src/main/java/minicraft/level/tile/HardRockTile.java

```diff
@@ -41,8 +41,9 @@ public boolean interact(Level level, int xt, int yt, Player player, Item item, D
 		if (item instanceof ToolItem) {
 			ToolItem tool = (ToolItem) item;
 			if (tool.type == ToolType.Pickaxe && tool.level == 4) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
-					hurt(level, xt, yt, random.nextInt(10) + (tool.level) * 5 + 10);
+				int dmg = random.nextInt(10) + tool.damage;
```

Comment:
tool.damage should be a private variable, and getters should be replaced with tool.getDamage() which includes randomness with a bound of 5.